main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateRouteNameValidator.java

```diff
@@ -107,33 +108,39 @@ private boolean areRoutesFromSameAgency(
   }
 
   /**
-   * Generate a key used to store {@code GtfsRoute} by `routes.route_long_name`.
+   * Generate an hash associated to `routes.route_long_name` and `routes.route_type`. This hash is
+   * used to interact with routeByLongName (variable defined in this class' validate method) to
+   * store and retrieve routes by short name.
    *
-   * @param route the {@code GtfsRoute} to generate the key from
-   * @return `routes.route_long_name`+`route.routeType`
+   * @param route the {@code GtfsRoute} to generate the hash from
+   * @return the hash associated to `routes.route_long_name` and `routes.route_type`.
    */
-  private String getRouteLongNameKey(GtfsRoute route) {
-    return route.routeLongName() + route.routeType();
+  private int getRouteLongNameRouteTypeHash(GtfsRoute route) {
```

Comment:
These method names could be shortened to remove `Route` to help with readability - for example, `getLongNameAndTypeHash()`